test/utility/parse_opts.c

```diff
@@ -0,0 +1,133 @@
+/*-
+ * Public Domain 2014-2016 MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "test_util.h"
+
+extern int   __wt_opterr;		/* if error message should be printed */
+extern int   __wt_optind;		/* index into parent argv vector */
+extern int   __wt_optopt;		/* character checked for validity */
+extern int   __wt_optreset;		/* reset getopt */
+extern char *__wt_optarg;		/* argument associated with option */
+
+/*
+ * testutil_parse_opts --
+ *    Parse command line options for a test case.
+ */
+int
+testutil_parse_opts(int argc, char *argv[], TEST_OPTS *opts)
+{
+	int ch;
+	size_t len;
+
+	opts->preserve = false;
+	opts->running = true;
+	opts->verbose = false;
+
+	if ((opts->progname = strrchr(argv[0], '/')) == NULL)
+		opts->progname = argv[0];
+	else
+		++opts->progname;
+
+	while ((ch =
+	    __wt_getopt(opts->progname,
+		argc, argv, "A:h:n:o:pR:T:t:vW:")) != EOF)
```

Comment:
There are a number of options used by other tests like `manydbs`, `readonly` and `recovery` that are not covered in this set.  We should try to change existing tests to use this infrastructure too.